docs/architecture/venice_architecture.md

```diff
@@ -0,0 +1,9 @@
+---
+layout: default
+title: Venice Architecture
+has_children: true
+permalink: /docs/architecture
```

Comment:
Hmm... I wonder if this is the right location for this doc. On one hand, I think it may be good to have a high-level architecture section. On the other hand, this particular page you're contributing goes pretty deep into the implementation details and I think it may be geared especially towards Venice developers/contributors, so I wonder if it should be housed under Developer Guides.

WDYT?

Reply:
I agree with that. I even thought about adding this to the user_guide and ended up creating a new section for it. However, the Dev guide sounds like a good place. It helps developers get up to speed in understanding the code structures and data flow. Let me create a subfolder `venice_internal` for this.